drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -407,27 +411,36 @@ def external_form_definition(identifier:)
       argument :identifier, String, required: true
     end
     def form_identifier(identifier:)
-      raise 'Access denied' unless current_user.can_configure_data_collection?
+      access_denied! unless current_user.can_configure_data_collection?
 
-      Hmis::Form::Definition.non_static.latest_versions.where(identifier: identifier).first
+      definition = Hmis::Form::Definition.non_static.latest_versions.where(identifier: identifier).first
+      raise 'not found' unless definition
+
+      ensure_form_role_permission(definition.role)
```

Comment:
Same comment here about the complexity not being strictly necessary

Reply:
Both sound good. I'm happy to revert to the simpler version.